main.py

```diff
@@ -131,6 +131,61 @@ def calculate_interest(self, rate: float) -> None:
         print("Debit accounts do not earn interest.")
 
 
+class SavingsAccount:
+    def __init__(self, account_number: str, balance: float = 0) -> None:
+        self.account_number = account_number
+        self.balance = balance
+        self.is_locked = False
+        self.transactions = []
+
+    def deposit(self, amount: float) -> None:
+        if not self.is_locked and amount > 0:
+            self.balance += amount
+            self.transactions.append(Transaction("Deposit", amount, datetime.datetime.now()))
+            print(f"Deposited ${amount}. New balance: ${self.balance}")
+        elif self.is_locked:
+            print("Account is locked. Cannot perform transactions.")
+        else:
+            print("Invalid deposit amount.")
+
+    def withdrawal(self, amount: float) -> None:
```

Comment:
If inheriting from `Account`, this could implement the `withdraw` abstract method. Granted there is an additional parameter there called `overdraft_protection` so it should be handled in the implementation.